### PR TITLE
UIAPPS-104 Link UI Extensions packages

### PR DESCRIPTION
**Pull request by joneshf-dd**
## Motivation

- Jira Issue: https://datadoghq.atlassian.net/browse/UIAPPS-104
- In https://github.com/DataDog/apps/pull/94, it turned out there was a lot of work just to make the packages link together properly. We pull that out so it's separated.

## Changes

- We make it so any package that starts with `@datadog/ui-extensions-` gets locked together with the same version. This took some extra work to make `changesets` still work the way we want. Hopefully if https://github.com/changesets/changesets/pull/690 merges, it'll mean we can remove a lot of this behavior.
    There's also a patch that we had to do to the `@changesets/assemble-release-plan` package that we should ask about.

## Releases

Choose one:

-   [x] No release is necessary.
        If you're only updating examples/documentation, this is likely what you want.
-   [ ] All packages that need a release have a changeset.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @datadog/create-app@1.0.3-canary.97.e634583.0
  npm install @datadog/framepost@0.3.1-canary.97.e634583.0
  npm install @datadog/ui-extensions-sdk@0.25.1-canary.97.e634583.0
  # or 
  yarn add @datadog/create-app@1.0.3-canary.97.e634583.0
  yarn add @datadog/framepost@0.3.1-canary.97.e634583.0
  yarn add @datadog/ui-extensions-sdk@0.25.1-canary.97.e634583.0
  ```
</details>

### CODE REVIEW

File: scripts/check-fixed-packages-ui-extensions.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * We do still want "linked" packages in our `.changeset/config.json`,
+ * since it means we'll always get everything on the same version.
+ * To that end,
+ * this check doesn't need to actually look at specific version numbers.
+ * We just have to catch packages that might be missed.
+ *
+ * The behavior we want is actually what's being proposed as "fixed" packages:
+ * https://github.com/changesets/changesets/pull/690.
+ * Until that feature is released,
+ * we have this check to make sure we release packages in lock-step.
```
Comment (nmuldavin) [inline scripts/check-fixed-packages-ui-extensions.js:25]: wow thank you for enforcing this in code and documenting so well. I tend to opt for manual enforcement of such things, which inevitably ends up not working

**Review by nmuldavin — APPROVED** [review]

**Comment by joneshf-dd** [comment]
thanks for the review!

**Comment by Andarist** [comment]
@joneshf-dd I've noticed that you are interested in the fixed mode that is being implemented for Changesets. I'm wondering if you have looked into the PR and if it matches your expectations or if maybe you have imagined the feature to work differently?

**Comment by joneshf-dd** [comment]
Hey, thanks for reaching out!

I think based on the description, and the linked conversations, it'll likely do what we want. But I haven't run any of that code yet, so I can't say with certainty. E.g. We found that the behavior of peer dependency version bumps didn't quite work out how we wanted and had to work around that. Even if there are things that don't work quite the way we want, if it lets us get rid of our ~60 lines of reactionary code in favor of a single line of proactive config, it'll be a win in my book.

No matter the case, I'm excited to try out the fixed mode. Can't wait for it to land!

**Pull request merged**